dfs/registry/utils.go

```diff
@@ -34,38 +35,68 @@ func GetRegistryImage(conn client.Connection, id string) (*registry.Image, error
 }
 
 // SetRegistryImage inserts a registry image into the coordinator index.
-func SetRegistryImage(conn client.Connection, rImage registry.Image) error {
+// Returns the UUID of the topmost layer of the specified image
+func SetRegistryImage(conn client.Connection, rImage registry.Image) (string, error) {
 	leaderpath := path.Join(zkregistryrepos, rImage.Library, rImage.Repo)
 	leadernode := &RegistryImageLeader{HostID: "master"}
 	if err := conn.CreateDir(leaderpath); err != nil && err != client.ErrNodeExists {
 		glog.Errorf("Could not create repo path %s: %s", leaderpath, err)
-		return err
+		return "", err
 	}
 	imagepath := path.Join(zkregistrytags, rImage.ID())
 	node := &RegistryImageNode{Image: rImage, PushedAt: time.Unix(0, 0)}
+	var retErr error
 	if err := conn.Create(imagepath, node); err == client.ErrNodeExists {
 		leader := conn.NewLeader(leaderpath, leadernode)
 		leaderDone := make(chan struct{})
 		defer close(leaderDone)
 		_, err := leader.TakeLead(leaderDone)
 		if err != nil {
-			return err
+			return "", err
 		}
 		defer leader.ReleaseLead()
 		if err := conn.Get(imagepath, node); err != nil {
```

Comment:
Agreed - but if the `Create` call here succeeds without error, then the `node` here would not be available to get the UUID from, and we'd end up needing a new connection to get it below